lib/katello/tasks/clean_backend_objects.rake

```diff
@@ -1,42 +1,77 @@
 namespace :katello do
   desc "Cleans backend objects (hosts) that are missing in one or more backend systems.  Run with COMMIT=true to commit changes."
   task :clean_backend_objects => ["environment", "check_ping"] do
-    def cleanup_hosts
-      Host.includes(:content_facet, :subscription_facet).find_each do |host|
-        if test_candlepin(host) || test_pulp(host)
-          print "Host #{host.id} #{host.name} #{host.subscription_facet.try(:uuid)} is partially missing.  Un-registering\n"
-          execute("Failed to delete host") { ForemanTasks.sync_task(::Actions::Katello::Host::Unregister, host) }
+    class BackendCleaner
+      def initialize
+        @candlepin_uuids = []
+        @pulp_uuids = []
+        @katello_candlepin_uuids = []
+        @katello_pulp_uuids = []
+      end
+
+      def populate!
+        @candlepin_uuids = Katello::Resources::Candlepin::Consumer.all_uuids
+        @katello_candlepin_uuids = Katello::Host::SubscriptionFacet.pluck(:uuid).compact
+
+        @pulp_uuids = ::Katello.pulp_server.extensions.consumer.retrieve(nil).map { |consumer| consumer['id'] }
```

Comment:
this .retrieve(nil) is kind of abusing this function.  Would you mind making this parameter optional in runcible?

Reply:
no need to update this, i think just making it optional (defaulting to nil) in runcible should ensure that this continues to work